### ISSUE TITLE
Min width / max width

### ISSUE THREAD

**Issue by WhyNotHugo** [ErikReider/SwayNotificationCenter#129]
Rather than specify an exact width for notification windows (e.g.: 500px), I'd prefer to define a range (e.g.: `minWidth: 300, maxWidth:500`). Only make the notification _wider_ than `minWidth`  if text won't fit in that size, and only grow up to `maxWidth`.

Does this make sense?

**Comment by ErikReider** [comment]
This would look strange with different sized notifications. Adding this would also break existing configs. Closing for now...

**Comment by WhyNotHugo** [comment]
> This would look strange with different sized notifications.

Quite true, that would not look right.

**Comment by noahgitsham** [comment]
> This would look strange with different sized notifications.

I disagree with this, and argue that some form of dynamic width would be quite useful. For example I came from dunst, and I used `notify-send "Calendar" "$(cal)"` as a simple way to check the calendar, and the notification automatically sized to the content. It would be great if there were a way to do this with swaync.

> Adding this would also break existing configs. Closing for now...

I am not familiar with JSON Schema, but could `notification-window-width` accept either a single value (so as to not break current configs) or a JSON object literal containing extra variables for dynamic width?
